### DISCUSSION TITLE
Disable automatic route generation of the custom API

### DISCUSSION THREAD

**Discussion by kratam** [nhost/nhost#41] [Ideas]
I'd like to be able to disable automatic route generation when using the custom API.
Similar to [this approach](https://medium.com/google-cloud/express-routing-with-google-cloud-functions-36fb55885c68), I'd like to handle the routes manually using express. This would allow me to use express middlewares for logging, error handling, etc.

**Comment by elitan** [comment]
Understandable. We're thinking of letting developers bring their own `Dockerfile` and thus do whatever they like. As long as the server listens on a `PORT` (env var). Similar to https://cloud.google.com/run

**Comment by sl1mpshady** [comment]
any plans to implement this?

**Reply by dbarrosop** [reply]
This was released some time ago:

https://nhost.io/product/run
